src/main/java/net/mcreator/ui/wysiwyg/WYSIWYG.java

```diff
@@ -304,8 +306,11 @@ private void drawGrid(Graphics2D g) {
 		ex = (int) Math.round(ex / 2.0);
 		ey = (int) Math.round(ey / 2.0);
 
-		if (positioningModeSettingWidth) {
+		if (positioningModeSettingWidth && !positioningModeSettingHeight) {
+			ow = Math.abs(ox - ex);
+		} else if (positioningModeSettingWidth && positioningModeSettingHeight) {
 			ow = Math.abs(ox - ex);
+			oh = Math.abs(oy - ey);

```

Comment:
You can simplify changes here so that this part looks similar to the **Resize** tool:
```java
		if (positioningModeSettingWidth) {
			ow = Math.abs(ox - ex);
			if (positioningModeSettingHeight)
				oh = Math.abs(oy - ey);
```